nexus/src/authz/api_resources.rs

```diff
@@ -191,6 +191,65 @@ impl AuthorizedResource for Fleet {
     }
 }
 
+/// ConsoleSessionList is a synthetic resource used for modeling who has access
+/// to create sessions.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct ConsoleSessionList;
```

Comment:
Why don't we need something like this to represent the ability to create the various other kinds of `FleetChild`?

Reply:
Good question.  There's one set of permissions and roles on Fleet, and it's basically written to be pretty locked-down.  You have to have Fleet "collaborator" to have the Fleet `create_child` permission.  But we want the "external-authenticator" to be able to create sessions without them having full write access to the Fleet.

A similar problem comes up with global images (in another PR).

Another approach would be to create more different actions on Fleet and have more roles that grant those permissions to do those actions.  The upside is we wouldn't need this synthetic resource.  I'm not sure it's clearer to do that, though.  It's a little funny to have these synthetic resources, but I think they neatly describe what's really going on (which is there's a collection -- albeit one not represented in the API -- that "internal-authenticator" has permission to create children in).  There's a more practical downside to creating a bunch more actions: right now, there's only one `authz::Action` enum with the union of all actions on all resources, so it would wind up growing a bunch of variants that aren't applicable for most resources.  I've been wondering if we could split this up.  The challenge is it's an argument to `OpContext::authorize()`.  Maybe it could be an associated type of the `AuthorizedResource` that you're doing a check on.  I haven't dug into this.

Reply:
Ok good, that's all helpful and matches the picture in my head to the extent I had one — that the synthetic resource is an alternative to adding a very narrow action to the enum for this purpose. I imagine @plotnick has an opinion about this.